### PR TITLE
gitignore patterns before globbing and 97

### PR DESCRIPTION
**Pull request by janat08**
closes https://github.com/xojs/xo/issues/235

### CODE REVIEW

**Comment by janat08** [comment]
it doesn't actually do anything outside of main api. Like hasmagic and generate glob tasks are modified. And the error checking code has to be duplicated.

**Comment by janat08** [comment]
also meant to fix #97

File: test.js [hunk 1]
```diff
@@ -6,13 +6,7 @@ import globby from '.';
...
 const cwd = process.cwd();
 const tmp = 'tmp';
-const fixture = [
-	'a.tmp',
-	'b.tmp',
-	'c.tmp',
-	'd.tmp',
-	'e.tmp'
-];
+const fixture = ['a.tmp', 'b.tmp', 'c.tmp', 'd.tmp', 'e.tmp'];
```
Comment (sindresorhus) [inline test.js]: Don't do unrelated changes. It's impossible to review this PR because of all the unrelated changes.

**Comment by sindresorhus** [comment]
> also meant to fix #97

That should be a separate PR.

**Comment by janat08** [comment]
I cherrypicked this branch, so I had to make another one to avoid linting changes.

**Comment by janat08** [comment]
#123
